### ISSUE TITLE
Slider Handle goes out of bounds when provided values lower/higher than min/max

### ISSUE THREAD

**Issue by cybae0804** [palantir/blueprint#4285]
#### Environment

- __Package version(s)__: 3.25.0
- __Operating System__: Ubuntu 18.04
- __Browser name and version__: Chrome 84

#### Code Sandbox

[sandbox](https://codesandbox.io/s/blueprint-sandbox-forked-gqlde) 

#### Steps to reproduce

1. Create a slider (also works for multislider)
2. Provide min/max props
3. Set the value to be greater than max value

#### Actual behavior

The handle goes out of bounds and floats to the side

#### Expected behavior

Should filter the value under the hood

#### Possible solution

`.filter(value => value > min] && value < max)` under the hood

This specifically came up because I'm trying to sync a slider with a d3 graph which zooms. I think everything works pretty nicely except for this out of bounds issue. I understand that this wouldn't happen under normal use of the slider where you would only be able to set the value prop of the slider by dragging and setting on the slider, but I also don't think this would be difficult/performance heavy.

**Comment by adidahiya** [comment]
I agree that we should add a logical guard against this edge case and prevent the component from getting in a bad state. But it should be implemented the same way as other components, as a validation check in `validateProps`. This check would throw an error or `console.error` when the supplied value is outside the min/max bounds, so you would need to do the "filtering" or clamping of the value on your own. I don't want the Blueprint component to do that, it's a bit too "magical" and breaks the easy-to-understand 1:1 correspondence of props -> component behavior.
